### ISSUE TITLE
SoftAssertions should have forward() or rethrow() method

### ISSUE THREAD

**Issue by Turbo87** [joel-costigliola/assertj-core#539]
My use case is the following: I have a folder with some files and would like to generate a unit test for every file. Since JUnit is not supporting anything like that I was thinking of using the `SoftAssertions` class instead. Unfortunately the test function itself is not under my control and uses JUnit assertions resulting in `AssertionErrors`.

I would like to catch those `AssertionErrors` and throw them again inside the `SoftAssertions` instance. I got two ideas for how this could work:

``` java
try {
    doTest(something);
} catch (AssertionError e) {
    softAssertions.rethrow(e);
}
```

of even simpler:

``` java
softAssertions.rethrow(() -> {
    doTest(something);
});
```

the only thing possible as of today looks like this:

``` java
try {
    doTest(something);
} catch (Throwable e) {
    softAssertions.assertThat(e).isNull();
}
```

but this is not printing the stacktrace or any other details of the caught exceptions.

**Comment by PascalSchumacher** [comment]
>My use case is the following: I have a folder with some files and would like to generate a unit test for >every file.

You could create a parameterized test and use the result of File.list() on the directory as parameter input. This way you get a test for every file in the directory.

**Comment by Turbo87** [comment]
@PascalSchumacher unfortunately I'm stuck on JUnit 3 for this due to the base class I have to use for this

**Comment by joel-costigliola** [comment]
@Turbo87 what @PascalSchumacher suggest would work, we could also add for 3.x something like:
```java
softAssertions.catchAssertionError(() -> {
    doTest(something);
})
```
Any other wording ideas welcome for `catchAssertionError` ...

**Comment by PascalSchumacher** [comment]
Joel fixed #864 in 2.9.0/3.9.0.

Imho we should just close this issue.

@joel-costigliola What do you think?
